python_modules/dagster/dagster/_core/events/__init__.py

```diff
@@ -1549,6 +1587,39 @@ def __new__(cls, asset_observation: AssetObservation):
         )
 
 
+class AssetFailedToMaterializeReason(Enum):
+    COMPUTE_FAILED = "COMPUTE_FAILED"
+    UPSTREAM_COMPUTE_FAILED = "UPSTREAM_COMPUTE_FAILED"
+
+
+@whitelist_for_serdes
+class AssetFailedToMaterializeData(
+    NamedTuple(
+        "AssetFailedToMaterializeData",
+        [
+            ("asset_key", AssetKey),
+            ("partition", Optional[str]),
+            ("reason", AssetFailedToMaterializeReason),
+            ("error", Optional[SerializableErrorInfo]),
```

Comment:
I could see us maybe wanting some additional fields in the future, like which upstream asset failed, but i think we can add that when needed